src/v/storage/log_manager.cc

```diff
@@ -86,14 +86,12 @@ void log_manager::trigger_housekeeping() {
 ss::future<> log_manager::stop() {
     _compaction_timer.cancel();
     _abort_source.request_abort();
-    return _open_gate.close()
-      .then([this] {
-          return ss::parallel_for_each(_logs, [](logs_type::value_type& entry) {
-              return entry.second->handle.close();
-          });
-      })
-      .then([this] { return _batch_cache.stop(); })
-      .then([this] { return async_clear_logs(); });
+    co_await _open_gate.close();
+    co_await ss::parallel_for_each(_logs, [](logs_type::value_type& entry) {
```

Comment:
I haven't looked at why, but `seastar::coroutine::parallel_for_each` exists, and maybe does so for some reason to avoid this non-coroutine version (but I can't think of a reason why the code here wouldn't work either..)

Reply:
Good tip -- the commit message on `include/seastar/coroutine/parallel_for_each.hh` explains it.  It's a minor effficiency improvement but seems appropriate to use in this case.